test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,454 @@
+
+        if let Some(permission_level) = value
+            .get("permission_level")
+            .and_then(|permission_level| permission_level.as_str())
+        {
+            if let Ok(parsed_permission_level) = permission_level.parse::<u32>() {
+                uattributes.permission_level = Some(parsed_permission_level);
+            } else {
+                error!("Error: Failed to parse permission_level as u32");
+            }
+        };
+
+        if let Some(commstatus_value) = value
+            .get("commstatus")
+            .and_then(|commstatus_value| commstatus_value.as_str())
+        {
+            uattributes.commstatus = Some(UCode::from_str(commstatus_value).unwrap().into());
+        } else {
+            error!("commstatus value is not string");
+        };
+
+        dbg!(" uattributes.commstatus: {:?}", uattributes.commstatus);
+
+        let mut ___reqid = UUID::new();
+        if let Some(resource) = value
+            .get("reqid")
+            .and_then(|resource| resource.get("lsb"))
+            .and_then(|resource| resource.as_str())
+        {
+            if let Ok(parsed_id) = resource.parse::<u64>() {
+                ___reqid.lsb = parsed_id;
+            } else {
+                error!("Error: Failed to parse _reqid_lsb as u64");
+            }
+        };
+
+        if let Some(resource) = value
+            .get("reqid")
+            .and_then(|resource| resource.get("msb"))
+            .and_then(|resource| resource.as_str())
+        {
+            if let Ok(parsed_id) = resource.parse::<u64>() {
+                ___reqid.msb = parsed_id;
+            } else {
+                dbg!("Error: Failed to parse _reqid_msb as u64");
+            }
+        };
+
+        uattributes.reqid = MessageField(Some(Box::new(___reqid)));
+
+        if let Some(token) = value.get("token") {
+            if let Some(token_str) = token.as_str() {
+                uattributes.token = Some(token_str.to_owned());
+            } else {
+                error!("Error: token is not a string");
+            }
+        };
+        if let Some(traceparent) = value
+            .get("traceparent")
+            .and_then(|traceparent| traceparent.as_str())
+        {
+            uattributes.traceparent = Some(traceparent.to_owned());
+        } else {
+            error!("Error: traceparent is not a string");
+        };
+
+        // special field //todo
+        let special_fields = SpecialFields::default();
+
+        if special_fields.ne(&SpecialFields::default()) {
+            uattributes.special_fields = special_fields;
+        }
+
+        Ok(WrapperUAttribute(uattributes))
+    }
+}
+#[derive(Default)]
+pub struct WrapperUPayload(pub UPayload);
+impl<'de> Deserialize<'de> for WrapperUPayload {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+        let mut upayload = UPayload::new();
```

Comment:
Same sorts of recommendations as [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1581473105):
> Suggest re-reading the recommendations I gave last time.
> 
> * try to avoid mutation
> * use the `let foo = ...` construct as suggested
> * construct each component (UAuthority, UEntity, UResource, UUri) one at a time e.g. as suggested [in this comment](https://github.com/eclipse-uprotocol/up-tck/pull/23#discussion_r1566166247)